src/test/java/seedu/sudohr/logic/commands/leave/DeleteEmployeeFromLeaveCommandTest.java

```diff
@@ -0,0 +1,413 @@
+import static seedu.sudohr.commons.util.CollectionUtil.requireAllNonNull;
+import static seedu.sudohr.logic.commands.CommandTestUtil.VALID_LEAVE_DATE_LEAVE_TYPE_1;
+import static seedu.sudohr.testutil.Assert.assertThrows;
+
+import java.nio.file.Path;
+import java.time.LocalDate;
+import java.util.function.Predicate;
+
+import org.junit.jupiter.api.Test;
+
+import javafx.collections.ObservableList;
+import javafx.collections.transformation.FilteredList;
+import seedu.sudohr.commons.core.GuiSettings;
+import seedu.sudohr.logic.commands.CommandResult;
+import seedu.sudohr.logic.commands.exceptions.CommandException;
+import seedu.sudohr.model.Model;
+import seedu.sudohr.model.ReadOnlySudoHr;
+import seedu.sudohr.model.ReadOnlyUserPrefs;
+import seedu.sudohr.model.SudoHr;
+import seedu.sudohr.model.department.Department;
+import seedu.sudohr.model.department.DepartmentName;
+import seedu.sudohr.model.employee.Employee;
+import seedu.sudohr.model.employee.Id;
+import seedu.sudohr.model.leave.Leave;
+import seedu.sudohr.model.leave.LeaveDate;
+import seedu.sudohr.testutil.TypicalEmployees;
+import seedu.sudohr.testutil.TypicalLeave;
+
+/**
+ * Deletes a employee from a specific leave in sudohr book.
+ */
+public class DeleteEmployeeFromLeaveCommandTest {
+
+    // Handle removing an employee
+    @Test
+    public void execute_employeeAcceptedByLeave_deleteSuccessful() throws CommandException {
+
+        ModelStubAcceptingEmployeeAdded modelStub = new ModelStubAcceptingEmployeeAdded();
+        modelStub.addEmployee(TypicalEmployees.ALICE);
+
+        new AddEmployeeToLeaveCommand(TypicalEmployees.ALICE_ID,
+                new LeaveDate(LocalDate.parse(VALID_LEAVE_DATE_LEAVE_TYPE_1))).execute(modelStub);
+
+        CommandResult commandResult = new DeleteEmployeeFromLeaveCommand(
+                TypicalEmployees.ALICE_ID,
+                new LeaveDate(LocalDate.parse(VALID_LEAVE_DATE_LEAVE_TYPE_1))).execute(modelStub);
+
+        assertEquals(String.format(DeleteEmployeeFromLeaveCommand.MESSAGE_SUCCESS,
+                TypicalEmployees.ALICE, TypicalLeave.LEAVE_TYPE_1),
+                commandResult.getFeedbackToUser());
+
+        assertFalse(modelStub.sudoHr.getLeave(new Leave(new LeaveDate(LocalDate.parse(
+                VALID_LEAVE_DATE_LEAVE_TYPE_1))))
+                .hasEmployee(TypicalEmployees.ALICE));
+    }
+
+    // Remove from nonexistent leave
+    @Test
+    public void execute_removeEmployeeFromNonExistentLeave_throwsCommandException() {
+        ModelStubAcceptingEmployeeAdded modelStub = new ModelStubAcceptingEmployeeAdded();
+
+        modelStub.addEmployee(TypicalEmployees.ALICE);
+
+        assertThrows(CommandException.class, DeleteEmployeeFromLeaveCommand.MESSAGE_INVALID_DATE, () ->
+                new DeleteEmployeeFromLeaveCommand(TypicalEmployees.ALICE_ID,
+                        new LeaveDate(LocalDate.parse(VALID_LEAVE_DATE_LEAVE_TYPE_1))).execute(modelStub));
+    }
+
+    // Remove a nonexistent employee
```

Comment:
is this test case accidentally swapped with the previous one? Benson is inside model stub right?

Reply:
This test case checks if deletion throw exception even if leave on a specific has been created for another user

Reply:
sorry slightly confused here.. so the employee exists but did not have leave on a given day. So here you are testing whether an error will be thrown if i remove a non-existent leave for an employee that exists?